tiledb/sm/cpp_api/array.h

```diff
@@ -566,23 +594,25 @@ class Array {
   }
 
   /**
-   * Reopens the array at a specific timestamp.
+   * This is a deprecated API.
```

Comment:
We need to properly deprecate this.

Reply:
@bekadavis9 I have given you bad information. I had told you to avoid the `TILEDB_DEPRECATED` macro because I thought it caused false positives just by importing the header.

The problem I was thinking of has to do with the C API. Using this macro on C APIs that are used in the implementation of the C++ API is what causes false positives. Let's add a `TILEDB_DEPRECATED` to all of the deprecated C++ APIs.

For the deprecated C APIs, you will need to either:
1. Ensure that they are not used in the C++ API, or
2. Refactor the C++ implementation to use non-deprecated APIs. If this is significant work, create a ticket to handle this later.